### ISSUE TITLE
[recharts 2.13] Function components cannot be given refs.

### ISSUE THREAD

**Issue by altay-oz** [brammitch/recharts-to-png#640]
Hi,

Thanks for your work on this project! I'm using some basic examples from Recharts and this project, but I'm encountering an issue where I'm unable to download the PNG file, and I’m seeing a ref error in the console; Function components cannot be given refs.

Here are the packages and their versions I'm using:
 
- React: 18.3.1
- Recharts: 2.13.0
- recharts-to-png: 2.3.2
- FileSaver: 2.0.5
 

Here's my code:

```javascript
import React, {useCallback} from 'react';
import { LineChart,
	 Line,
	 XAxis,
	 YAxis,
	 CartesianGrid,
	 Tooltip,
	 Legend,
	 ResponsiveContainer } from 'recharts';

import { useCurrentPng } from "recharts-to-png";
import FileSaver from "file-saver";

function App() {

  const data = [
    { name: 'Page A', uv: 4000, pv: 2400, amt: 2400 },
    { name: 'Page B', uv: 3000, pv: 1398, amt: 2210 },
    { name: 'Page C', uv: 2000, pv: 9800, amt: 2290 },
    { name: 'Page D', uv: 2780, pv: 3908, amt: 2000 },
    { name: 'Page E', uv: 1890, pv: 4800, amt: 2181 },
    { name: 'Page F', uv: 2390, pv: 3800, amt: 2500 },
    { name: 'Page G', uv: 3490, pv: 4300, amt: 2100 },
  ];

  const [getPng, { ref, isLoading }] = useCurrentPng();

  const handleDownload = useCallback(async () => {
    const png = await getPng();

    if (png) {
      FileSaver.saveAs(png, "myChart.png");
    }
  }, [getPng]);

  return (
    <>
      <ResponsiveContainer width="100%" height={300}>
	<LineChart
          width={500}
          height={300}
          data={data}
          margin={{
            top: 5,
            right: 30,
            left: 20,
            bottom: 5,
          }}
	  ref={ref}
	>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip />
          <Legend />
          <Line type="monotone" dataKey="pv" stroke="#8884d8" activeDot={{ r: 8 }} />
          <Line type="monotone" dataKey="uv" stroke="#82ca9d" />
	</LineChart>
      </ResponsiveContainer>
      <button onClick={handleDownload} >
	{isLoading ? 'Downloading...' : 'Download Chart'}
      </button>
    </>
  );

}

export default App;
```

Any guidance on resolving this would be greatly appreciated! Thank you.

**Comment by ckifer** [comment]
Note that removal of `ref` from recharts was not intentional 😅

**Comment by brammitch** [comment]
> Note that removal of `ref` from recharts was not intentional 😅

Noted 😁

**Comment by altay-oz** [comment]
@brammitch, thanks a lot. Now by switching useCurrentPng to useGenerateImage and wrapping the chart with a <div> made my code work again :)

**Comment by ckifer** [comment]
Should be fixed in 2.13.2

**Comment by brammitch** [comment]
Verified examples are working with recharts@2.13.2.
